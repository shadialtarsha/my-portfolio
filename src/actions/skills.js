// import fetch from 'node-fetch';

export const setSkills = (skills = []) => ({
  type: 'SET_SKILLS',
  skills,
});

export const startSetSkills = () => dispatch => {
  const url = 'https://shadi-my-portfolio.herokuapp.com/api/skills';
  return fetch(url)
    .then(response => response.json())
    .then(data => dispatch(setSkills(data.skills)))
    .catch(e => console.log(e));
};
