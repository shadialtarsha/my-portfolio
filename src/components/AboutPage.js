import React from 'react';
import { NavLink } from 'react-router-dom';
import Menu from './Menu';
import SocialMedia from './SocialMedia';

const AboutPage = () => (
  <div>
    <Menu />
    <div className="page">
      <div className="centered" id="about">
        <div className="wrapper">
          <div className="about-container">
            <div className="avatar">
              <div className="img" />
            </div>
            <div>
              <h3>Hello, My name is Shadi Al-Tarsha.</h3>
              <p>
                <span className="special-word">I am a Software developer</span>, ambitious, self-motivated, team worker,
                and a big believer in the self-taught approach.
              </p>
              <p>
                My favorite programming language is JavaScript, but Java was my best friend for the first 2 years of my
                career.
              </p>
              <p>
                Check out my projects in the{' '}
                <NavLink className="special-link" to="/portfolio">
                  portfolio page
                </NavLink>.
              </p>
              <p>
                If you want to make something exsiting together, feel free to email me at{' '}
                <span className="special-word">work@shadialtarsha.com</span> or fill a form in the{' '}
                <NavLink className="special-link" to="/contact">
                  contact page
                </NavLink>.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <NavLink to="/skills" className="arrow" id="arrow-right">
      <div id="right-arrow-link">skills</div>
      <div className="lines">
        <div id="right-upper" className="line" />
        <div id="right-lower" className="line" />
      </div>
    </NavLink>
    <SocialMedia />
  </div>
);

export default AboutPage;
