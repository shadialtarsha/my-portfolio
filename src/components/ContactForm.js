import React, { Component } from 'react';

export class ContactForm extends Component {
  state = {
    name: '',
    email: '',
    message: '',
  };

  onNameChange = e => {
    const { target: { value } } = e;
    this.setState(() => ({ name: value }));
    const nameLabel = this.nameLabelRef.current;
    if (nameLabel && document.activeElement === this.nameInputRef.current) {
      if (this.validateName(value)) {
        nameLabel.style.color = '#009688';
      } else {
        nameLabel.style.color = '#c60202';
      }
    }
  };

  onEmailChange = e => {
    const { target: { value } } = e;
    this.setState(() => ({ email: value }));
    const emailLabel = this.emailLabelRef.current;
    if (emailLabel && document.activeElement === this.emailInputlRef.current) {
      if (this.validateEmail(value)) {
        emailLabel.style.color = '#009688';
      } else {
        emailLabel.style.color = '#c60202';
      }
    }
  };

  onMessageChange = e => {
    const { target: { value } } = e;
    this.setState(() => ({ message: value }));
    const messageLabel = this.messageLabelRef.current;
    if (messageLabel && document.activeElement === this.messageTextareaRef.current) {
      if (this.validateMessage(value)) {
        messageLabel.style.color = '#009688';
      } else {
        messageLabel.style.color = '#c60202';
      }
    }
  };

  onSubmit = e => {
    e.preventDefault();
    this.nameLabelRef.current.style.removeProperty('color');
    this.emailLabelRef.current.style.removeProperty('color');
    this.messageLabelRef.current.style.removeProperty('color');
    this.setState(() => ({
      name: '',
      email: '',
      message: '',
    }));
  };

  validateName = name => name.length > 1;
  validateMessage = message => message.length > 1;
  validateEmail = email => {
    const regx = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return regx.test(email);
  };

  enableButton = () =>
    this.validateName(this.state.name) &&
    this.validateEmail(this.state.email) &&
    this.validateMessage(this.state.message);

  nameInputRef = React.createRef();
  emailInputlRef = React.createRef();
  messageTextareaRef = React.createRef();
  nameLabelRef = React.createRef();
  emailLabelRef = React.createRef();
  messageLabelRef = React.createRef();

  render() {
    return (
      <form onSubmit={this.onSubmit} className="contact-form">
        <div>
          <input
            id="name"
            type="text"
            value={this.state.name}
            placeholder="Enter name"
            name="name"
            ref={this.nameInputRef}
            onChange={this.onNameChange}
          />
          <label htmlFor="name" ref={this.nameLabelRef}>
            <i className="fas fa-user" />
          </label>
        </div>
        <div>
          <input
            id="email"
            type="email"
            value={this.state.email}
            placeholder="Enter email"
            name="email"
            ref={this.emailInputlRef}
            onChange={this.onEmailChange}
          />
          <label htmlFor="email" ref={this.emailLabelRef}>
            <i className="fas fa-at" />
          </label>
        </div>
        <div id="textarea-container">
          <textarea
            id="message"
            value={this.state.message}
            name="message"
            placeholder="Enter message"
            rows="6"
            ref={this.messageTextareaRef}
            onChange={this.onMessageChange}
          />
          <label htmlFor="message" ref={this.messageLabelRef}>
            <i className="fas fa-envelope" />
          </label>
        </div>
        <div className="buttons-container">
          <button id="submit" type="submit" disabled={!this.enableButton()}>
            Send a message
          </button>
          <button
            id="download"
            onClick={() => {
              window.open('http://localhost:3333/portfolio/resume', '_blank');
            }}
          >
            My resume as pdf
          </button>
        </div>
      </form>
    );
  }
}

export default ContactForm;
