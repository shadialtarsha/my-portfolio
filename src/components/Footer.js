import React from 'react';
import { NavLink } from 'react-router-dom';

const Footer = () => (
  <footer>
    <h1>Let's talk</h1>
    <p>Wanna get in touch or talk about a project?</p>
    <p>Feel free to contact me via email at work@shadialtarsha.com</p>
    <p>
      or drop a line in the form at the <NavLink to="/contact">contact page</NavLink>
    </p>
  </footer>
);

export default Footer;
