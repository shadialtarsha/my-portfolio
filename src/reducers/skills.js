const skillsReducerDefaultState = [];

const skillsReducer = (state = skillsReducerDefaultState, action) => {
  switch (action.type) {
    case 'SET_SKILLS':
      return action.skills;
    default:
      return state;
  }
};

export default skillsReducer;
