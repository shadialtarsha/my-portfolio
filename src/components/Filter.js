import React from 'react';
import { connect } from 'react-redux';
import { TransitionGroup, CSSTransition } from 'react-transition-group';
import { setFilter } from '../actions/filter';
import filterProjects from '../selectors/filterProjects';

const tags = [
  'Show All',
  'JavaScript',
  'React',
  'Redux',
  'SASS',
  'JAVA',
  'Firebase',
  'Android',
  'Front-end',
  'FullStack',
  'NodeJs',
  'ExpressJs',
  'RESTFUL',
  'Mongodb',
  'JEST',
];

export class Filter extends React.Component {
  state = {
    currentFilter: 'SHOW ALL',
  };

  componentDidMount() {
    this.props.setFilter(tags[0]);
  }

  setActive = e => {
    const previousActive = this.filterItemsRef.current.querySelector('.active');
    if (previousActive !== e.currentTarget) {
      previousActive.classList.remove('active');
      e.currentTarget.classList.add('active');
      const currentFilter = e.currentTarget.innerText;
      this.setState(() => ({ currentFilter }));
      document.querySelector('.splash-layer').classList.add('show-splash-layer');
      document.querySelector('.splash-layer span').classList.add('show-span');
      setTimeout(() => {
        this.props.setFilter(currentFilter);
        document.querySelector('#projects-status').innerText =
          this.state.currentFilter === 'SHOW ALL'
            ? 'Showing all projects.'
            : `Showing ${this.props.numberOfProject} filtered by ${this.props.filter}.`;
        document.querySelector('.splash-layer').classList.remove('show-splash-layer');
        document.querySelector('.splash-layer span').classList.remove('show-span');
      }, 1500);
    }
  };

  filterItemsRef = React.createRef();

  render() {
    return (
      <div>
        <div className="splash-layer">
          <span>
            {this.state.currentFilter === 'SHOW ALL' ? 'SHOW ALL' : `FILTERING ${this.state.currentFilter} PROJECTS`}
          </span>
        </div>
        <ul className="filter-items" ref={this.filterItemsRef}>
          {tags.map(tag => {
            let item = (
              <li key={tag} onClick={this.setActive}>
                {tag}
              </li>
            );
            if (tag === 'Show All') {
              item = (
                <li key={tag} className="active" onClick={this.setActive}>
                  {tag}
                </li>
              );
            }
            return item;
          })}
        </ul>
        <p id="projects-status">Showing all projects.</p>
      </div>
    );
  }
}

const mapStateToProps = ({ filter, projects }) => ({
  filter,
  numberOfProject: filterProjects(projects, filter).length,
});

const mapDispatchToProps = dispatch => ({
  setFilter: filter => dispatch(setFilter(filter)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Filter);
