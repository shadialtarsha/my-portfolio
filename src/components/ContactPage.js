import React from 'react';
import { NavLink } from 'react-router-dom';
import Menu from './Menu';
import ContactForm from './ContactForm';
import SocialMedia from './SocialMedia';

const ContactPage = () => (
  <div>
    <Menu />
    <ContactForm />
    <SocialMedia />
    {/* <NavLink to="/portfolio">portfolio</NavLink> */}
  </div>
);

export default ContactPage;
