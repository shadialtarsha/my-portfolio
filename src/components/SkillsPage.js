import React from 'react';
import { connect } from 'react-redux';
import Header from './Header';
import SkillItem from './SkillItem';
import Footer from './Footer';

export const SkillsPage = ({ skills }) => (
  <div>
    <Header />
    <div>
      <header>
        <h1>My Skills</h1>
        <p>These are my major skills as a Software engineer</p>
        {skills.map(skill => <SkillItem key={skill._id} skill={skill} />)}
      </header>
    </div>
    <Footer />
  </div>
);

const mapStateToProps = state => ({
  skills: state.skills,
});

export default connect(mapStateToProps)(SkillsPage);
