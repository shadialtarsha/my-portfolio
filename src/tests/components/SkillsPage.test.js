import React from 'react';
import { shallow } from 'enzyme';
import { SkillsPage } from '../../components/SkillsPage';
import skills from '../fixtures/skills';

test('should render skillsPage correctly', () => {
  const wrapper = shallow(<SkillsPage skills={skills} />);
  expect(wrapper).toMatchSnapshot();
});
