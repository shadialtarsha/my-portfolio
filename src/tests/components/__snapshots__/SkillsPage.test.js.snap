// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render skillsPage correctly 1`] = `
<div>
  <Header />
  <div>
    <header>
      <h1>
        My Skills
      </h1>
      <p>
        These are my major skills as a Software engineer
      </p>
      <SkillItem
        key="html1"
        skill={
          Object {
            "_id": "html1",
            "name": "HTML",
            "percentage": "75%",
          }
        }
      />
      <SkillItem
        key="css2"
        skill={
          Object {
            "_id": "css2",
            "name": "CSS",
            "percentage": "75%",
          }
        }
      />
      <SkillItem
        key="javascript3"
        skill={
          Object {
            "_id": "javascript3",
            "name": "JavaScript",
            "percentage": "80%",
          }
        }
      />
      <SkillItem
        key="react4"
        skill={
          Object {
            "_id": "react4",
            "name": "React",
            "percentage": "70%",
          }
        }
      />
      <SkillItem
        key="redux5"
        skill={
          Object {
            "_id": "redux5",
            "name": "Redux",
            "percentage": "70%",
          }
        }
      />
      <SkillItem
        key="unittest6"
        skill={
          Object {
            "_id": "unittest6",
            "name": "Unit Test",
            "percentage": "75%",
          }
        }
      />
      <SkillItem
        key="nodejs7"
        skill={
          Object {
            "_id": "nodejs7",
            "name": "NodeJS",
            "percentage": "75%",
          }
        }
      />
      <SkillItem
        key="express8"
        skill={
          Object {
            "_id": "express8",
            "name": "ExpressJS",
            "percentage": "70%",
          }
        }
      />
      <SkillItem
        key="mongodb9"
        skill={
          Object {
            "_id": "mongodb9",
            "name": "MongoDB",
            "percentage": "65%",
          }
        }
      />
      <SkillItem
        key="java10"
        skill={
          Object {
            "_id": "java10",
            "name": "JAVA",
            "percentage": "75%",
          }
        }
      />
      <SkillItem
        key="android11"
        skill={
          Object {
            "_id": "android11",
            "name": "Android",
            "percentage": "70%",
          }
        }
      />
      <SkillItem
        key="firebase12"
        skill={
          Object {
            "_id": "firebase12",
            "name": "Firebase",
            "percentage": "75%",
          }
        }
      />
      <SkillItem
        key="sql13"
        skill={
          Object {
            "_id": "sql13",
            "name": "SQL",
            "percentage": "65%",
          }
        }
      />
      <SkillItem
        key="restfulapi14"
        skill={
          Object {
            "_id": "restfulapi14",
            "name": "RESTFUL API",
            "percentage": "75%",
          }
        }
      />
    </header>
  </div>
  <Footer />
</div>
`;
